pkg/builtins/bitwise_test.go

```diff
@@ -0,0 +1,146 @@
+	mem := memory.NewMemorySegmentManager()
+	mem.AddSegment()
+	mem.Memory.Insert(memory.NewRelocatable(0, 5), memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(10)))
+	mem.Memory.Insert(memory.NewRelocatable(0, 6), memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(12)))
+
+	builtin := builtins.NewBitwiseBuiltinRunner(true)
+
+	address := memory.NewRelocatable(0, 7)
+
+	result, err := builtin.DeduceMemoryCell(address, &mem.Memory)
+	if err != nil {
+		t.Errorf("Test failed with error: %v", err)
+	}
+	expected := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(8))
+
+	if *result != *expected {
+		t.Errorf("TestDeduceMemoryCellBitwiseForPresetMemoryValidAnd Failed, expected %v, got %v", expected, result)
+	}
+
+}
+
+func TestDeduceMemoryCellBitwiseForPresetMemoryValidXor(t *testing.T) {
+	mem := memory.NewMemorySegmentManager()
+	mem.AddSegment()
+	mem.Memory.Insert(memory.NewRelocatable(0, 5), memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(10)))
+	mem.Memory.Insert(memory.NewRelocatable(0, 6), memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(12)))
+
+	builtin := builtins.NewBitwiseBuiltinRunner(true)
+
+	address := memory.NewRelocatable(0, 8)
+
+	result, err := builtin.DeduceMemoryCell(address, &mem.Memory)
+
+	if err != nil {
+		t.Errorf("Test failed with error: %v", err)
+	}
+	expected := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(6))
+
+	if *result != *expected {
+		t.Errorf("TestDeduceMemoryCellBitwiseForPresetMemoryValidAnd Failed, expected %v, got %v", expected, result)
+	}
+
+}
+
+func TestDeduceMemoryCellBitwiseForPresetMemoryValidOr(t *testing.T) {
+	mem := memory.NewMemorySegmentManager()
+	mem.AddSegment()
+	mem.AddSegment()
+	mem.Memory.Insert(memory.NewRelocatable(0, 5), memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(10)))
+	mem.Memory.Insert(memory.NewRelocatable(0, 6), memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(12)))
+
+	builtin := builtins.NewBitwiseBuiltinRunner(true)
+
+	address := memory.NewRelocatable(0, 9)
+
+	result, err := builtin.DeduceMemoryCell(address, &mem.Memory)
+
+	if err != nil {
+		t.Errorf("Test failed with error: %v", err)
+	}
+	expected := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(14))
+
+	if *result != *expected {
+		t.Errorf("TestDeduceMemoryCellBitwiseForPresetMemoryValidAnd Failed, expected %v, got %v", expected, result)
+	}
+
+}
+
+func TestDeduceMemoryCellBitwiseForPresetMemoryIncorrectOffset(t *testing.T) {
+	mem := memory.NewMemorySegmentManager()
+	mem.AddSegment()
+	rel1 := memory.NewRelocatable(0, 3)
+	rel2 := memory.NewRelocatable(0, 4)
+	rel3 := memory.NewRelocatable(0, 5)
+
+	m1 := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(10))
+	m2 := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(12))
+	m3 := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(0))
+
+	mem.Memory.Insert(rel1, m1)
+	mem.Memory.Insert(rel2, m2)
+	mem.Memory.Insert(rel3, m3)
```

Comment:
I think this line can be removed, because `Memory.Get` returns 0 by default

Reply:
It doesn't return 0 by default, it returns a zero value when it fails, if we don't insert the 0, Memory.Get will still fail (aka return (0, false))

Reply:
In fact it fails, we are returning this: 

`value, ok := m.data[addr]

	if !ok {
		return nil, errors.New("Memory Get: Value not found")
	}
`

But in the tests that position was not being loaded from memory so it did not fail, I add again the values